### PR TITLE
[KOGITO-1336] - Improve the exposed HTTP port configuration for Kogito Runtime Services

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Many thanks for submiting your Pull Request :heart:! 

Jira Issue : https://issues.redhat.com/browse/KOGITO-1336

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: pkg/infrastructure/kogito_service.go [hunk 1]
```diff
@@ -18,6 +18,15 @@ import (
...
 	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"strconv"
+)
+
+const (
+	// HTTPPortEnvVar ... Env variable to define port on which service will listen internally
+	HTTPPortEnvVar = "HTTP_PORT"
```
Comment (spolti) [inline pkg/infrastructure/kogito_service.go:29]: Are you going to configure the images to accept this environment?
notice that data-index images uses the KOGITO_DATA_INDEX_HTTP_PORT env, and for management console there is no env exposed so far.

Comment (spolti) [reply pkg/infrastructure/kogito_service.go:29]: if this env would be used to set the port, we'll need to modify the management console and data-index images as well.

Comment (spolti) [reply pkg/infrastructure/kogito_service.go:29]: nevermind, I didn't see the other PR, sry lol

**Comment by sutaakar** [comment]
Will check once Zanini's comment is addressed.

File: pkg/infrastructure/services/resources.go [hunk 2]
```diff
@@ -295,3 +300,35 @@ func (s *serviceDeployer) getComparator() compare.MapComparator {
...
+			container.ReadinessProbe.HTTPGet != nil {
+			container.ReadinessProbe.HTTPGet.Port = intstr.IntOrString{IntVal: httpPort}
+			container.LivenessProbe.HTTPGet.Port = intstr.IntOrString{IntVal: httpPort}
+		}
+	}
+}
+
+// defineHTTPPort will define on which port the service should be listening to. To set it use httpPort cr parameter.
+// defaults to 8080
+func defineHTTPPort(kogitoService v1alpha1.KogitoService) int32 {
```
[the remark this answers is not in the document]
Comment (vaibhavjainwiz) [reply pkg/infrastructure/services/resources.go:325]: done

File: pkg/infrastructure/services/resources.go [hunk 3]
```diff
@@ -295,3 +300,35 @@ func (s *serviceDeployer) getComparator() compare.MapComparator {
 
 	return compare.MapComparator{Comparator: resourceComparator}
 }
+
+// configureHTTPPort ... Set HTTP port in env variable of given container
+func configureHTTPPort(deployment *appsv1.Deployment, kogitoService v1alpha1.KogitoService) {
+	if len(deployment.Spec.Template.Spec.Containers) > 0 {
+		container := &deployment.Spec.Template.Spec.Containers[0]
+		httpPort := defineHTTPPort(kogitoService)
```
[the remark this answers is not in the document]
Comment (vaibhavjainwiz) [reply pkg/infrastructure/services/resources.go:308]: done

File: pkg/infrastructure/services/resources.go [hunk 4]
```diff
@@ -129,6 +133,7 @@ func (s *serviceDeployer) applyDeploymentCustomizations(deployment *appsv1.Deplo
 			return err
 		}
 	}
+	configureHTTPPort(deployment, s.instance)
```
[the remark this answers is not in the document]
Comment (vaibhavjainwiz) [reply pkg/infrastructure/services/resources.go]: done

File: pkg/infrastructure/services/resources.go [hunk 5]
```diff
@@ -295,3 +300,35 @@ func (s *serviceDeployer) getComparator() compare.MapComparator {
 
 	return compare.MapComparator{Comparator: resourceComparator}
 }
+
+// configureHTTPPort ... Set HTTP port in env variable of given container
```
[the remark this answers is not in the document]
Comment (vaibhavjainwiz) [reply pkg/infrastructure/services/resources.go]: done

File: pkg/framework/env_var.go [hunk 6]
```diff
@@ -107,6 +107,17 @@ func SetEnvVar(key, value string, container *corev1.Container) {
 	container.Env = append(container.Env, corev1.EnvVar{Name: key, Value: value})
 }
 
+// AppendEnvVar will update or add the environment variable
+func AppendEnvVar(key, value string, envs []corev1.EnvVar) []corev1.EnvVar {
```
Comment (sutaakar) [inline pkg/framework/env_var.go:111]: Seems to duplicate EnvOverride function above.

Comment (sutaakar) [reply pkg/framework/env_var.go:111]: Although the EnvOverride is using EnvVar as parameter.
You can construct EnvVar parameter here and call EnvOverride to reuse the implementation.

**Comment by ricardozanini** [comment]
> In KogitoApp the custom port is applied just in env variable, but for example not in probes.

It's a bug then. Since we are deprecating it, won't worth fixing it,  wdyt?

**Comment by sutaakar** [comment]
>  It's a bug then. Since we are deprecating it, won't worth fixing it, wdyt?
Fine for me.

File: pkg/infrastructure/services/deployment.go [hunk 7]
```diff
@@ -20,11 +20,14 @@ import (
...
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strconv"
 )
 
 const (
 	portName      = "http"
 	singleReplica = int32(1)
+	// HTTPPortEnvKey Env variable to define port on which service will listen internally
+	HTTPPortEnvKey = "HTTP_PORT"
```
Comment (ricardozanini) [inline pkg/infrastructure/services/deployment.go:30]: Could you please reconsider exporting this constant?

File: pkg/infrastructure/services/deployment.go [hunk 8]
```diff
@@ -74,3 +79,23 @@ func createRequiredDeployment(service v1alpha1.KogitoService, resolvedImage stri
...
+	if httpPort < 1 {
+		log.Debugf("HTTPPort not set, returning default http port.")
+		return framework.DefaultExposedPort
+	}
+	log.Debugf("HTTPPort is set, returning port number %i", httpPort)
+	return httpPort
+}
+
+// SetHTTPPortInEnvVar will update or add the environment variable into the given kogito service
+func SetHTTPPortInEnvVar(httpPort int32, kogitoService v1alpha1.KogitoService) {
```
Comment (ricardozanini) [inline pkg/infrastructure/services/deployment.go:97]: I believe this function also can be private.

File: pkg/framework/env_var.go (hunk 6 above)
Comment (vaibhavjainwiz) [reply pkg/framework/env_var.go:111]: done

File: pkg/infrastructure/services/deployment.go (hunk 8 above)
Comment (vaibhavjainwiz) [reply pkg/infrastructure/services/deployment.go:97]: done

File: pkg/infrastructure/services/deployment.go (hunk 7 above)
Comment (vaibhavjainwiz) [reply pkg/infrastructure/services/deployment.go:30]: Currently this const is refered from `deployment_config.go` file. We can change its scope once we remove kogito app

**Comment by spolti** [comment]
let's reschedule it to 0.12, its too close from release and it can bring new issues.

**Comment by ricardozanini** [comment]
So we need a PR to bump the version so @vaibhavjainwiz can rebase to update the CRD accordingly.

**Pull request merged**